agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileCache.java

```diff
@@ -35,6 +38,16 @@
    */
   private final Queue<String> persistedFilesCache = new ConcurrentLinkedDeque<>();
 
+  public LocalFileCache(File folder) {
+    Collection<File> files = FileUtils.listFiles(folder, new String[] {"trn"}, false);
+    // existing files are not older than 48 hours and need to get added to the queue to be
+    // re-processed.
+    // this will avoid data loss in the case of app crashes and restarts.
+    for (File file : files) {
+      persistedFilesCache.add(file.getName());
+    }
```

Comment:
good catch, these were not being read previously?

Reply:
correct.